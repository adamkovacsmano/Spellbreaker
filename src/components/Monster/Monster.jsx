import React, { Component } from "react";
import styles from "./Monster.module.scss";

class Monster extends Component{
  state = {};

  render(){
    return(
      <div className={styles.monsterContainer} style={{cursor: "pointer"}}>
        <h1>{this.props.name}</h1>
        <img src={this.props.img} alt="monster-pic"></img>
        <section className={styles.content}>
          <p>{this.props.description}</p>
          <tr className={styles.attributes}>
            <div>
              <tr>Combat expertise:</tr>
              <td>{this.props.combatexp}</td>
            </div>
            <div>
              <tr>HP:</tr>
              <td>{this.props.hp}</td>
            </div>
          </tr>
        </section>
      </div>
    );
  }
}

export default Monster;