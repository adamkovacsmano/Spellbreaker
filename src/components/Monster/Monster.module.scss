.monsterContainer {
  width: 350px;
  border: 4px solid black;
  border-radius: 10px;
  box-sizing: border-box;
  padding: 10px;
  text-align: center;
  background-image: url(../../Images/cardbackground.jpg);
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;

  h1 {
    font-size: 1.5em;
    color: rgb(255, 60, 60);
  }

  p {
    border: 1px solid white;
    padding: 5px;
    color: white;
  }

  img {
    width: 300px;
    height: 200px;
    border: 1px solid black;
    object-fit: cover;
  }

  table {
    width: 100%;
    padding: 5px;
  }
}

.attributes {
  display: flex;
  justify-content: space-between;
  background-color: black;
  color: white;
  padding: 5px 10px;
  border: 1px solid white;

  div{
    display: flex;
  }

  tr {
    padding-right: 5px;
  }

  td {
    color: rgb(255, 60, 60);
  }
}