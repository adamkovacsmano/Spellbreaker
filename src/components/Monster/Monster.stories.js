import React from "react";
import Monster from "./Monster";

export default {
  title: "Monster"
};

export const monster = () => <Monster />;