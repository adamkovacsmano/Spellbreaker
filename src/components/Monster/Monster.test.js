import React from "react";
import { shallow } from "enzyme";
import Monster from "./Monster";

describe("CreatePost tests", () => {
  let monster;
  beforeEach(() => {
    monster = shallow(<Monster />);
  });

  test("should match the snapshot", () => {
    expect(monster).toMatchSnapshot();
  });
});
