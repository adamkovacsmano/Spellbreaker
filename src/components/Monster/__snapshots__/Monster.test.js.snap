// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost tests should match the snapshot 1`] = `
<div
  className="monsterContainer"
>
  <h1 />
  <img
    alt="monster-pic"
  />
  <section>
    <p />
    <tr
      className="attributes"
    >
      <div>
        <tr>
          Combat expertise:
        </tr>
        <td />
      </div>
      <div>
        <tr>
          HP:
        </tr>
        <td />
      </div>
    </tr>
  </section>
</div>
`;
